apps/web-desmos/src/screens/proposal_details/components/overview/components/metadata_loader/index.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from 'react';
+import Loading from '@/components/loading';
+
+interface MetadataLoaderProps {
+  metadata: string;
+}
+
+// Checks if a string is a valid URL
+const isValidUrl = (url: string) => {
+  const pattern = /^(ftp|http|https|ipfs):\/\/[^ "]+$/;
+  return pattern.test(url);
+};
```

Comment:
These generic functions are duplicated in a few apps. They seem generic to me so I think in future cases I would add them into the base and then import them from there (even if not used in the base), but just idea